mypy/checker.py

```diff
@@ -1619,11 +1619,15 @@ def check_compatibility(self, name: str, base1: TypeInfo,
         first = base1[name]
         second = base2[name]
         first_type = first.type
-        if first_type is None and isinstance(first.node, FuncBase):
-            first_type = self.function_type(first.node)
+        if first_type is None:
+            method = get_callable(first.node)
+            if method:
+                first_type = self.function_type(method)
         second_type = second.type
-        if second_type is None and isinstance(second.node, FuncBase):
-            second_type = self.function_type(second.node)
+        if second_type is None:
+            method = get_callable(second.node)
+            if method:
+                second_type = self.function_type(method)
```

Comment:
I am not sure why you need this change, but please take into account that there are _three_ places in `checker.py` that check Liskov: one here (for "horizontal" overriding in multiple inheritance), one for variables, and one for methods etc. (There is some code duplication, I know our code is not perfect.) So please double check if you need to update other two places.